interview-questions.md

```diff
@@ -8,50 +8,56 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is the difference between a parameter and an argument?
 
-Your answer:
+Your answer: A parameter is the placeholder name inside a function.
+    An argument is data passed into a function when invoked/called.
 
-Researched answer:
+Researched answer: A parament is a variable placehoder that is shown inside the functions parenthesis or (). It can be used again and again in the function once in the parenthesis or ().  
+An argument though is the value assigned to the parameter or the inputs of the function, and there has to be an equal number of parameters as arguments.  
 
 2. The JavaScript built in method .map() takes predefined parameters. What are they? Which are required and which are optional?
 
-Your answer:
+Your answer: Ok Value, Index, and the Array are required for .map() and I think that the Index is not always required as long as you have the array and the value.  
 
-Researched answer:
+Researched answer: the .map() method takes Value, Index, and Array as parameters, and the value is required to hold the value of the element. The Index and Array of the HOG are not required paramenters, both are optional for the .map(), if you know the value you will have the index as it goes in order from left to right starting count at zero.  
 
 3. What is the difference between map and filter?
 
-Your answer:
+Your answer: The main difference between .map() & .filter() is that .filter() will always try to return an array shorter than the original array. 
 
-Researched answer:
+Researched answer: .map() and .filter() both work on HOF however .map() after iterating through the array will do something to each item and  return a new array of the same length similar to a for loop.  .filter() also iterates through the array but returns a new shorter array and it requires an eval that returns a boolean type logic value which is how it filters the original array. 
 
 4. What is iteration?
 
-Your answer:
+Your answer: An iteration is the output of a For Loop Function. 
```

Comment:
Make sure to expand on this as much as possible.  One-line answers do not give you enough chance to demonstrate your knowledge thoroughly.